pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -278,7 +278,7 @@ func (s *NodeController) PopulateAddresses(nodes *api.NodeList) (*api.NodeList,
 				glog.Errorf("error getting instance ip address for %s: %v", node.Name, err)
 			} else {
 				address := api.NodeAddress{Type: api.NodeLegacyHostIP, Address: hostIP.String()}
-				node.Status.Addresses = append(node.Status.Addresses, address)
+				api.AddToNodeAddresses(&node.Status.Addresses, address)
```

Comment:
We can override the address here:  node.Status.Addresses = []api.Nodeaddress{address}